### ISSUE TITLE
CSVFormatDetector is capturing a quote escape when it shouldn't.

### ISSUE THREAD

**Issue by stefan-zh** [uniVocity/univocity-parsers#105]
I found that in a CSV that has the following ASCII characters, the quote escape detection would give a wrong result:
```
"2","1-"\n
```
The quote escape would be set to `-` even though that's an expected part of the value.

I looked at the code and found this line in `CSVFormatDetector.java` https://github.com/uniVocity/univocity-parsers/blob/master/src/main/java/com/univocity/parsers/csv/CsvFormatDetector.java#L86
```java
if (Character.isLetterOrDigit(next) || next <= ' ') { //no special characters after quote, might be escaping
   //special character before (potentially) closing quote, might be an escape
   char prev = characters[i - 1];
   if (!Character.isLetterOrDigit(prev)) {
      increment(escape, prev);
   }
}
```
I think that a row delimiter after a quote that should match a previously open quote should not be escaped.

**Comment by jbax** [comment]
A single row is a very restricted input. The algorithm builds statistics using a few rows of a random input  as a sample and does its best to guess the format using some heuristics. It can't be 100% accurate because the problem is non-deterministic.

**Comment by stefan-zh** [comment]
Thanks for the feedback @jbax. I understand the problem is very difficult and also found that in the documentation it has been specified that `the detection process is not guaranteed to discover the correct quote escape.` Unfortunately, I ran into a use case where automatic detection for quote escaping resulted in wrongly parsed feed. The reason was that input such as `"abc-"\n` was used to determine `-` was an escape character for the quote following. Anyway, I decided to disable the automatic detection of quotes and instead explicitly specify the quotes and quote-escape character according tho this thread: http://stackoverflow.com/questions/17808511/properly-escape-a-double-quote-in-csv
